### PR TITLE
Attempt to fix VM detection on Windows

### LINKED ISSUE #3738: Windows 11 with Hyper-V installed detected as VM.
When running Webots in Windows 11 with Hyper-V installed, the WbSysInfo::isVirtualMachine will return true even when Webots is running on the Host system.

This cause a rendering performance degradation due to excessive glFlush calls.

**Steps to Reproduce**
Run Webots with gdb on a Windows 11 machine with Hyper-V installed.
put breakpoint in isVirtualMachine and check the result.

**Expected behavior**
Webots correctly detects that it is not running in a VM.

**System**
 - Windows 11 x64 with Hyper-V installed
 - Nvidia RTX 3090 24GB.

Now what is strange is that the 31st bit of the ECX register in the result of the CPUID EAX=1 call is set.
This indicates the presence of an HyperVisor.
See [CPUID](https://en.wikipedia.org/wiki/CPUID#Calling_CPUID) information.

Now this might well be a Windows 11 ONLY issue.
I can't check that on Windows 10.

### PR DESCRIPTION
**Pull request by omichel**
Might fix #3738. Taken from https://stackoverflow.com/a/66974571/810268

### CODE REVIEW

**Comment by llessieux** [comment]
I'll check that as soon as I can.

**Comment by llessieux** [comment]
By the way, I have a fix for the Windows 11 detection too if you want. Same file.
It is just an extra build number check to switch between Windows 10 and 11.

File: src/webots/core/WbSysInfo.cpp [hunk 1]
```diff
@@ -368,7 +368,7 @@ bool WbSysInfo::isVirtualMachine() {
   __get_cpuid(0x1, &eax, &ebx, &ecx, &edx);
   if (!(ecx & (1 << 31))) {
     virtualMachine = 1;
```
Comment (llessieux) [inline src/webots/core/WbSysInfo.cpp:368]: I think that the virtualMachine should be set to 0 in this case.

**Review by llessieux — CHANGES_REQUESTED** [review]

**Comment by llessieux** [comment]
That's the change I have tested for the Windows 11 detection.

    case QSysInfo::WV_WINDOWS10:  // Or Windows 11
    {
      QString version("Windows 10");
      QString kernel_version = QSysInfo::kernelVersion();
      QRegExp rx("[.]");
      QStringList list = kernel_version.split(rx, Qt::SkipEmptyParts);
      if (list.size() == 3) {
        int build_number = list[2].toInt();
        if (build_number >= 2200) {
          version = "Windows 11";
        }
      }
      sysInfo.append(version);
      break;

**Comment by omichel** [comment]
That looks good. May I ask you to commit it in there?

**Comment by llessieux** [comment]
> 
> 
> That looks good. May I ask you to commit it in there?

Sure. But that will have to wait until tomorrow ;) Getting late here :)

Ok I can confirm that your fix seems to be working here.

**Comment by llessieux** [comment]
Ok Windows 11 detection added.

**Comment by omichel** [comment]
It doesn't work quite well as on my Windows 10 machine, it says "Windows 11". The list returned in `kernelVersion` is `("10", "0", "19043")` on my machine, which obviously leads to "Windows 11".

**Comment by llessieux** [comment]
I think that there is a zero missing :) It should be 22000.

**Comment by llessieux** [comment]
You can check it also by looking at the version number of NtosKernel.exe in C:\Windows\System32.

**Review by llessieux — APPROVED** [review]

**Pull request merged**